core/src/ancestor_hashes_service.rs

```diff
@@ -136,6 +136,40 @@ pub struct AncestorHashesService {
     thread_hdls: Vec<JoinHandle<()>>,
 }
 
+enum PacketThresholdUpdate {
+    Increase,
+    Decrease,
+}
+
+struct DynamicPacketToProcessThreshold {
+    max_packets: usize,
+}
+
+impl Default for DynamicPacketToProcessThreshold {
+    fn default() -> Self {
+        Self {
+            max_packets: Self::DEFAULT_MAX_PACKETS,
+        }
+    }
+}
+
+impl DynamicPacketToProcessThreshold {
+    const DEFAULT_MAX_PACKETS: usize = 1024;
+    const TIME_THRESHOLD_MS: u64 = 1000;
+    const FACTOR: f64 = 0.9;
```

Comment:
Let's avoid floats.  This can just be an integer percentage and retain the original math. Switch to saturating math for bonus points!

Reply:
Fixed.